src/test/java/org/eclipse/uprotocol/communication/TestUTransport.java

```diff
@@ -26,6 +33,24 @@ public class TestUTransport implements UTransport {
     private final UUri mSource;
 
     public UMessage buildResponse(UMessage request) {
+        // If the request is a subscribe or unsubscribe request, return the appropriate response
+        if (request.getAttributes().getSink().getUeId() == 0) {
+            if (request.getAttributes().getSink().getResourceId() == 1) {
+                try {
+                    SubscriptionRequest subscriptionRequest = SubscriptionRequest.parseFrom(request.getPayload());
+                    SubscriptionResponse subResponse = SubscriptionResponse.newBuilder()
+                        .setTopic(subscriptionRequest.getTopic())
+                        .build();
+                    return UMessageBuilder.response(request.getAttributes()).build(UPayload.pack(subResponse));
+                } catch (InvalidProtocolBufferException e) {
+                    return UMessageBuilder.response(request.getAttributes()).build(
+                        UPayload.pack(UnsubscribeResponse.newBuilder().build()));
+                }
+            } else {
+                return UMessageBuilder.response(request.getAttributes()).build(
+                    UPayload.pack(UnsubscribeResponse.newBuilder().build()));
```

Comment:
I think we should copy request id to response reqId here

Reply:
the request id is in the UAtributes that is fed to the UMessageBuilder.response() static method

Reply:
The requestID is copied from the UMessageBuilder.response() static method when it is passed the request.getAttributes()